day12/writex.py

```diff
@@ -0,0 +1,51 @@
+def writeX(matrix):
+    rows = set()
+    cols = set()
+
+    height = len(matrix[0])
+    width = len(matrix)
+
+    for i in xrange(height):
+        for j in xrange(width):
+            if matrix[i][j] == 'x':
+                rows.add(i)
+                cols.add(j)
+
+    for row in rows:
+        for j in xrange(width):
+            if matrix[row][j] != 'x':
```

Comment:
interesting point about the fewer machine instructions though, I never think about that